modules/calib3d/src/calibration.cpp

```diff
@@ -1725,15 +1728,15 @@ static double cvCalibrateCamera2Internal( const CvMat* objectPoints,
             CvMat _dpdf = cvMat(_Ji.colRange(0, 2));
             CvMat _dpdc = cvMat(_Ji.colRange(2, 4));
             CvMat _dpdk = cvMat(_Ji.colRange(4, NINTRINSIC));
-            CvMat _dpdo = cvMat(_Jo.colRange(0, ni * 3));
+            CvMat _dpdo = _Jo.empty() ? CvMat() : cvMat(_Jo.colRange(0, ni * 3));
```

Comment:
Looks like all `_dpd*` variables are used only inside `cvProjectPoints2Internal` call.
Wouldn't it be better to create corresponding `cvMat` structures inside the call directly?

Reply:
i let these cvMat initialisation to respect the original the code.
you are right, i'll update this.

Reply:
Since `cvProjectPoints2Internal` takes **cvMat*** 
Making the creation of _dp* matrices at call site makes gcc angry : 

> ./src/github/opencv/modules/calib3d/src/calibration.cpp:1732:61: error: taking address of temporary [-fpermissive]

i'll move them under the if statement to reduce the scope of these variables.